cli/resolver.rs

```diff
@@ -116,14 +119,51 @@ impl Resolver for CliGraphResolver {
     specifier: &str,
     referrer: &ModuleSpecifier,
   ) -> Result<ModuleSpecifier, AnyError> {
-    // TODO(bartlomieju): actually use `maybe_package_json_deps` here, once
-    // deno_graph refactors and upgrades land.
     if let Some(import_map) = &self.maybe_import_map {
-      import_map
+      return import_map
         .resolve(specifier, referrer)
-        .map_err(|err| err.into())
-    } else {
-      deno_graph::resolve_import(specifier, referrer).map_err(|err| err.into())
+        .map_err(|err| err.into());
+    }
+
+    if !self.no_npm {
+      if let Ok(_node_builtin_module) = resolve_builtin_node_module(specifier) {
+        return Ok(
+          ModuleSpecifier::parse(&format!(
+            "node:{}",
+            specifier.strip_prefix("node:").unwrap_or(specifier)
+          ))
+          .unwrap(),
+        );
+      }
+    }
+
+    if let Some(deps) = self.maybe_package_json_deps.as_ref() {
+      if let Some(req) = deps.get(specifier) {
+        return Ok(ModuleSpecifier::parse(&format!("npm:{req}")).unwrap());
+      }
+    }
+
+    match deno_graph::resolve_import(specifier, referrer)
+      .map_err(|err| err.into())
+    {
+      Ok(resolved) => Ok(resolved),
+      Err(err) => {
+        if !self.no_npm {
+          if let Some(local_node_modules_path) =
+            self.maybe_local_node_modules_path.as_ref()
```

Comment:
Maybe we should store this as a `ModuleSpecifier` instead. Then we wouldn't have to do `to_file_path()`